src/vaults/VaultV2.sol

```diff
@@ -0,0 +1,109 @@
+        // Profit must not be unlocking
+        require(block.timestamp >= lastHarvest + LOCK_INTERVAL, "BV: profit unlocking");
+
+        // Get the Vault's current total strategy holdings.
+        uint256 oldTotalStrategyHoldings = totalStrategyHoldings;
+
+        // Used to store the new total strategy holdings after harvesting.
+        uint256 newTotalStrategyHoldings = oldTotalStrategyHoldings;
+
+        // Used to store the total profit accrued by the strategies.
+        uint256 totalProfitAccrued;
+
+        for (uint256 i = 0; i < strategyList.length; i = uncheckedInc(i)) {
+            // Get the strategy at the current index.
+            Strategy strategy = strategyList[i];
+
+            // Ignore inactive (removed) strategies
+            if (!strategies[strategy].isActive) {
+                continue;
+            }
+
+            // Get the strategy's previous and current balance.
+            uint232 balanceLastHarvest = strategies[strategy].balance;
+            uint256 balanceThisHarvest = strategy.totalLockedValue();
+
+            // Update the strategy's stored balance.
+            strategies[strategy].balance = uint232(balanceThisHarvest);
+
+            // Increase/decrease newTotalStrategyHoldings based on the profit/loss registered.
+            // We cannot wrap the subtraction in parenthesis as it would underflow if the strategy had a loss.
+            newTotalStrategyHoldings = newTotalStrategyHoldings + balanceThisHarvest - balanceLastHarvest;
+
+            unchecked {
+                // Update the total profit accrued while counting losses as zero profit.
+                // Cannot overflow as we already increased total holdings without reverting.
+                totalProfitAccrued += balanceThisHarvest > balanceLastHarvest
+                    ? balanceThisHarvest - balanceLastHarvest // Profits since last harvest.
+                    : 0; // If the strategy registered a net loss we don't have any new profit.
+            }
+        }
+
+        if (totalProfitAccrued > 0) {
+            uint256 perfFee = totalProfitAccrued.mulDiv(performanceFeeBps, MAX_BPS, MathUpgradeable.Rounding.Up);
+            accumulatedPerformanceFee += uint128(perfFee);
+            totalProfitAccrued -= perfFee;
+            newTotalStrategyHoldings -= perfFee;
+        }
+
+        // Update max unlocked profit based on any remaining locked profit plus new profit.
+        maxLockedProfit = uint128(lockedProfit() + totalProfitAccrued);
+
+        // Set strategy holdings to our new total.
+        totalStrategyHoldings = newTotalStrategyHoldings;
+
+        // Assess fees (using old lastHarvest) and update the last harvest timestamp.
+        _assessFees();
+        lastHarvest = uint128(block.timestamp);
+
+        emit Harvest(msg.sender, strategyList);
+    }
+
+    function withdrawPerformanceFee() external virtual onlyGovernance {
```

Comment:
any possibility of user withdrawing all the assets before withdrawing performance fees. 

Also need to calculate TVL reducing the performance fees.

Reply:
> any possibility of user withdrawing all the assets before withdrawing performance fees.

I don't think so. The user's will always be withdrawing using the official reduced tvl which does not include performance fees. E.g. If we start with $1 and gain $1, given a performance fee of 50% the tvl used for minting/burning shares will be $1.5. Even if all users withdraw, there should still be $0.5 left over. Slippage can affect this though.

Reply:
> Also need to calculate TVL reducing the performance fees.

Are you talking about when the vault takes a loss? Right now governance keeps unclaimed performance fees even if there is a loss afterwards.